api/reviews.go

```diff
@@ -50,10 +51,12 @@ func (api *API) addReviewHandler(writer http.ResponseWriter, request *http.Reque
 	}
 
 	review.ID = uuid.NewV4().String()
+	review.BookID = bookID
 	review.Links.Self = fmt.Sprintf("/books/%s/reviews/%s", bookID, review.ID)
 	review.Links.Book = fmt.Sprintf("/books/%s", bookID)
+	review.LastUpdated = time.Now().UTC()
```

Comment:
This is all common logic for creating a new review object which will be duplicated each time someone needs to create a new review object. It also places a burden on the caller to always have to set it up.  Consider adding a constructor function to the models package to encapsulate this e.g.:

```
func NewReview(bookID string) Review {
    ...
}
```
Whenever someone needs to create a new review they would simple call:
````
review := models.NewReview(bookID)
````
Another argument to use this approach is that the `review.ID` feels like it should be owned by the review model and not something the caller decides. If the format or method of generating it were ever to change it would be difficult to enforce, if its handled inside the package we could update it however we want and it wouldn't affect the caller.

Again I know I sound pedantic but applying encapsulation and enforcing clear separation of concerns makes things easier to change later on - its much easier to avoid a tangled mess or dependencies. My 2 cents anyway.

Reply:
Thanks, I was feeling uneasy about the way I was assigning the IDs